### ISSUE TITLE
Ping notification when it is someones' turn

### ISSUE THREAD

**Issue by The10DrForever** [pocc/bga_discord#42]
Would it at all be possible for this bot to send an @ message to a player when it is their turn via discord. I know the website has push notifications however, it can still be difficult when newer players forget to check if it is their turn. A simple @ message from the bot would be very helpful.

Thanks.

**Comment by pocc** [comment]
This is a good idea, but will require the bot to poll table statuses unless there's some way to receive updates from BGA itself.

**Comment by The10DrForever** [comment]
> This is a good idea, but will require the bot to poll table statuses unless there's some way to receive updates from BGA itself.

Yeah I am not sure about how all of this works. I have little to no coding knowledge but I just wanted to see if it could even be done.

**Comment by mavit** [comment]
It should be possible in principle for the bot to subscribe to the push notifications and bridge them to Discord.

**Comment by pocc** [comment]
Unless we have access to the user's browser, I don't think we can receive push notifications. It might be possible to do something with selenium to login and receive these? But then you'd have a bunch of active selenium sessions.

**Comment by mavit** [comment]
If someone was keen enough to work on it, if I understand correctly it would be technically possible for the bot to host its own [endpoints for the Push API](https://w3c.github.io/push-api/#dfn-push-endpoint) (firewalls and sufficiently stable IP address allowing).

**Comment by pocc** [comment]
Oh ok. So it sounds like it's technically feasible, provided BGA's push notifications are up to spec.

**Comment by molenear** [comment]
All your table statuses are in a single file.  I wrote a shell script that uses my PHPSESSID to get my /gameinprogress page, query the embedded JSON blob for turn info, convert the BGA user IDs to Discord IDs (manual table), and write all that to a file.  My bot uses that file to create the @mention announcements.  The script and mentions are triggered via a bot command, so the data displayed is always fresh.  One command (turns) bothers everyone with a waiting turn, and another command (myturns) only shows that person's turns.

One stat I'd hoped for is missing: the turn timer.  This would make it possible to notify users when they have a new turn to take.  I have plans to calculate this in the script but haven't been motivated.

Push would absolutely be more efficient, but this glorified screen scraping works for us.  I can share the relevant code if anyone is interested.
